### PR TITLE
COFF: add ARMNT and ARM64 support ##bin

### PR DESCRIPTION
**Pull request by pelijah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

Dummy loader for COFF objects was added to avoid issues with alignment in ARM binaries.

Related issue: #7310.

Simple example: `pdf` output of function `buffer_putflush()` https://github.com/radareorg/sdb/blob/master/src/buffer.c#L53-L58  
```asm
[0x000000d0]> pdf @ sym.buffer_putflush
/ 68: sym.buffer_putflush ();
|           ; var int16_t var_0h @ sp+0x0
|           ; var int16_t var_4h @ sp+0x4
|           ; var int16_t var_8h @ sp+0x8
|           ; var int16_t var_18h @ sp+0x18
|           ; var int16_t var_1ch @ sp+0x1c
|           ; var int16_t var_20h @ sp+0x20
|           0x00000218      0fb4           push {r0, r1, r2, r3}
|           0x0000021a      2de90048       push.w {fp, lr}
|           0x0000021e      eb46           mov fp, sp
|           0x00000220      84b0           sub sp, 0x10
|           0x00000222      0698           ldr r0, [var_18h]
|           0x00000224      fff772ff       bl sym.buffer_flush         ; RELOC 32 buffer_flush @ 0x0000010c
|           0x00000228      0190           str r0, [var_4h]
|           0x0000022a      019b           ldr r3, [var_4h]
|           0x0000022c      002b           cmp r3, 0
|       ,=< 0x0000022e      02d1           bne sym._LN2_2
|       |   0x00000230      0023           movs r3, 0
|       |   0x00000232      0093           str r3, [sp]
|      ,==< 0x00000234      0ce0           b sym._LN1_2
|      ||   ;-- $LN2:
|      |`-> 0x00000236      089b           ldr r3, [var_20h]
|      |    0x00000238      079a           ldr r2, [var_1ch]
|      |    0x0000023a      0699           ldr r1, [var_18h]
|      |    0x0000023c      0c31           adds r1, 0xc
|      |    0x0000023e      0968           ldr r1, [r1]
|      |    0x00000240      0698           ldr r0, [var_18h]
|      |    0x00000242      1030           adds r0, 0x10
|      |    0x00000244      0068           ldr r0, [r0]
|      |    0x00000246      00f009f8       bl sym.allwrite             ; RELOC 32 allwrite @ 0x0000025c
|      |    0x0000024a      0290           str r0, [var_8h]
|      |    0x0000024c      029b           ldr r3, [var_8h]
|      |    0x0000024e      0093           str r3, [sp]
|      |    ; CODE XREF from sym.buffer_putflush @ 0x234
|      |    ;-- $LN1:
|      `--> 0x00000250      0098           ldr r0, [sp]
|           0x00000252      04b0           add sp, 0x10
|           0x00000254      5df804bb       ldr fp, [sp], 4
\           0x00000258      5df814fb       ldr pc, [sp], 0x14
```
COFF object itself [buffer.zip](https://github.com/radareorg/radare2/files/5398022/buffer.zip)

### CODE REVIEW

**Review by trufae — COMMENTED** [review]
Yay! One more target supporting arm64

How’s that nany x86 tests have different offsets now? Lgtm by reading the code

**Comment by pelijah** [comment]
> 
> How’s that nany x86 tests have different offsets now? Lgtm by reading the code

Dummy loader for COFF objects was added to avoid issues with alignment in ARM binaries.

File: test/db/formats/coff
Comment (radare) [inline test/db/formats/coff:76]: these offsets look wrong to me but I didnt looked in depth. can you confirm?

File: libr/bin/p/bin_coff.c
Comment (radare) [inline libr/bin/p/bin_coff.c:561]: we can add strdup_arm() :DDD, jk

File: libr/bin/p/bin_coff.c
Comment (radare) [inline libr/bin/p/bin_coff.c:483]: uhm, i bet there's an r_io api to do that. cc @condret

File: libr/bin/format/coff/coff_specs.h
Comment (radare) [inline libr/bin/format/coff/coff_specs.h:156]: i prefer space indents to avoid those problems when the first char of the line is not a tab

**Review by radare — CHANGES_REQUESTED** [review]
Also, add ##bin in the PR title

File: test/db/formats/coff
Comment (pelijah) [reply test/db/formats/coff:76]: `LEA`
0x06 + 0x03 = 0x09
0x0d + 0x03 = 0x10
`CALL`
0x1b + 0x01 = 0x1c
0x22 + 0x01 = 0x23
```asm
0x00000000         41b900000000  mov r9d, 0
0x00000006       4c8d0523000000  lea r8, [rip + 0x23]
0x0000000d       488d152b000000  lea rdx, [rip + 0x2b]
0x00000014       48c7c100000000  mov rcx, 0
0x0000001b           e830000000  call 0x50
0x00000020                 89c1  mov ecx, eax
0x00000022           e831000000  call 0x58
```

File: libr/bin/format/coff/coff_specs.h
Comment (pelijah) [reply libr/bin/format/coff/coff_specs.h:156]: Every line of the file with `#define` contains tabs. Should I format them all?

Comment (trufae) [reply libr/bin/format/coff/coff_specs.h:156]: yep

Comment (XVilka) [reply libr/bin/format/coff/coff_specs.h:156]: I think it's better to avoid reformatting changes in unrelated PR. It can be done separately.

Comment (pelijah) [reply libr/bin/format/coff/coff_specs.h:156]: @radare @XVilka  So what is your final decision?

Comment (XVilka) [reply libr/bin/format/coff/coff_specs.h:156]: @pelijah don't do formatting in this PR. Lets merge it as is.

**Review by XVilka — APPROVED** [review]

**Pull request merged**
